### DISCUSSION TITLE
Scoped Containers with TypeOrm -> Cannot get Connection with name "default"

### DISCUSSION THREAD

**Discussion by MWhite-22** [MichalLytek/type-graphql#939] [Q&A] [answered]
**Describe the Bug**

When using scoped containers with ApolloServer, TypeGraphQL, and TypeORM, no resolver works. Everything is great for global containers, just not scoped. The error I get back is 

```
"Cannot get Connection with name "default" from the ConnectionManager. 
Make sure you have created the connection and called "useContainer(Container)" in your application before establishing a connection and importing any entity into TypeORM.",
```

**To Reproduce**

```ts
/*...other imports*/
import { ApolloServer } from 'apollo-server-express';
import { buildSchema, ResolverData } from 'type-graphql';
import { getConnectionManager, useContainer } from 'typeorm';
import { Container as TypeOrmContainer } from 'typeorm-typedi-extensions';
import Container from 'typedi';

useContainer(TypeOrmContainer);

async function main() {
	// ============================================================
	// 			SETUP
	// ============================================================
	const app = Express();
	app.set('trust proxy', 1);

	// ============================================================
	// 			DB
	// ============================================================
	try {
		const dbConnection = await DB.connect();
		console.log('TypeORM Connected:', dbConnection.isConnected);
	} catch (e) {
		console.warn('TypeORM Connection Error');
		console.log(e);
		process.exit(1);
	}

	// ============================================================
	// 			APOLLO SERVER
	// ============================================================
	const apolloServer = new ApolloServer({
		schema: await buildSchema({
			resolvers: [path.join(process.cwd(), 'build/modules/**/resolver*.js')],
			container: ({ context }: ResolverData<CTX>) => context.scopedContainer, //GRAB SCOPE CONTAINER HERE
			validate: { validationError: { target: false, value: false } },
		}),
		context: ({ req, res }): CTX => {
			const requestId: string = cuid();
			const scopedContainer = Container.of(requestId); //CREATE SCOPE CONTAINER HERE

			const context: CTX = {
				requestId,
				req,
				res,
				session: req.session,
				scopedContainer,
			};

			scopedContainer.set<CTX>('CTX', context);

			return context;
		},
		playground: { settings: { 'request.credentials': 'include' } },
	});

	apolloServer.applyMiddleware({
		app,
		cors: { origin: ENV.FRONTEND.URL, credentials: true },
	});

	// ============================================================
	// 			START SERVER
	// ============================================================
	app.listen(ENV.PORT, () => console.log(`Server started on http://localhost:${ENV.PORT}/graphql`));
}
```

Everything builds as expected, but try to query any resolver class that has @InjectRepository() or anything TypeORM related, and GQL returns the above error message. 

I have a suspicion this is a "typeorm-typedi-extensions" issue with the type-di changes from 0.8 to 0.10, but if there's any help you could provide on fixes that would be amazing.

Full repo [HERE](https://github.com/Xzas22/TypeORM_ScopeContainer_Issue)

**Expected Behavior**

Connect to DB and query repository as intended. Everything works without scopes, and scoped containers work great without typeorm, just no go on the combo application.

**Environment (please complete the following information):**
- OS: Mac
- Node 15.12.0
- Package version 1.1.1
- TypeScript version 4.3.4</div>

**Comment by MichalLytek** [comment]
Moving to discussion as it's not TypeGraphQL issue.

**Comment by MichalLytek** [comment]
TypeGraphQL has nothing to do with the DI containers. It just calls `.get()` and that's all.

> I have a suspicion this is a "typeorm-typedi-extensions" 

Ask on their repo how to configure TypeDI, TypeORM and scoped containers.

**Reply by MWhite-22** [reply]
Yup, already on it. You are just significantly more active and, given the docs frequent connection to TypeORM, I was wondering if you had any recommendations on correct configuration. 

Ill wait for a reply from the typeDI team. Thanks.

**Comment by MWhite-22** [comment, marked as answer]
FYI I Found the error. Its in the way typeorm-typedi-extensions sets up the ConnectionManager in TypeDI.

Deep dive into results here: https://github.com/typeorm/typeorm-typedi-extensions/issues/51
